packages/elements/src/components/ino-datepicker/validator.ts

```diff
@@ -0,0 +1,193 @@
+    if (this._isDisabled) {
+      return true;
+    }
+
+    if (!value && this._isRequired) {
+      return false;
+    }
+
+    if (!this.validateFormat(value)) {
+      return false;
+    }
+
+    if (this._minDate && !this.validateMin(value)) {
+      return false;
+    }
+
+    if (this._maxDate && !this.validateMax(value)) {
+      return false;
+    }
+
+    return true;
+  };
+
+  private validateMin = (value: string): boolean =>
+    this._isRanged
+      ? this.validateMinRange(value)
+      : this.validateMinSingle(value);
+
+  private validateMinSingle = (value: string): boolean => {
+    const parsedDate: Date = this.parseDate(value);
+    return (
+      isEqual(parsedDate, this._minDate) || isAfter(parsedDate, this._minDate)
+    );
+  };
+
+  private validateMinRange = (value: string): boolean => {
+    return !Validator.convertToRangeArray(value)
+      .map(this.validateMinSingle)
+      .includes(false);
+  };
+
+  private validateMax = (value: string): boolean =>
+    this._isRanged
+      ? this.validateMaxRange(value)
+      : this.validateMaxSingle(value);
+
+  private validateMaxSingle = (value: string): boolean => {
+    const parsedDate: Date = this.parseDate(value);
+    return (
+      isEqual(parsedDate, this._maxDate) || isBefore(parsedDate, this._maxDate)
+    );
+  };
+
+  private validateMaxRange = (value: string): boolean => {
+    return !Validator.convertToRangeArray(value)
+      .map(this.validateMaxRange)
+      .includes(false);
+  };
+
+  private validateFormat = (value: string): boolean =>
+    this._isRanged
+      ? this.validateFormatRange(value)
+      : this.validateFormatSingle(value);
+
+  private validateFormatSingle = (value: string): boolean => {
+    try {
+      const parsedDate: Date = this.parseDate(value);
+      const formattedDate: string = flatpickr.formatDate(
+        parsedDate,
+        this._dateFormat
+      );
+      return formattedDate == value;
+    } catch (_) {
+      throw new Error(
+        `Value (${value}) could not be parsed. Make sure to provide a date in the following format: ${this._dateFormat}`
+      );
+    }
+  };
+
+  private validateFormatRange = (value: string): boolean => {
+    return !Validator.convertToRangeArray(value)
+      .map(this.validateFormatSingle)
+      .includes(false);
+  };
+
+  private parseDate = (value: string): Date => {
+    const parsedDate = flatpickr.parseDate(value, this._dateFormat);
+
+    if (!parsedDate) {
+      throw new Error(`Could not parse Date ${value}`);
+    }
+
+    return parsedDate;
+  };
+
+  set isDisabled(value: boolean) {
+    this._isDisabled = value;
+  }
+
+  set isRequired(value: boolean) {
+    this._isRequired = value;
+  }
+
+  set dateFormat(value: string) {
+    this._dateFormat = value;
+  }
+
+  set minDate(value: string | undefined) {
+    if (!value) {
+      this._minDate = undefined;
+      return;
+    }
+
+    try {
+      this._minDate = this.parseDate(value);
```

Comment:
The issue described below produces unnecessary console errors if `minDates` are initialized asynchronously. Maybe the `minDate`as well as `maxDate` only be parsed if it is not falsy (null|undefined|'"') otherwise it is set to undefined